docs/enhancements/tenancy-based-auth.md

```diff
@@ -0,0 +1,83 @@
+| Delete InfraEnv| For bound InfraEnv, allowed for the owner and for each user that has remove access<br>for the bound cluster's subscription. For unbound InfraEnv, allowed for the owner                                               |
+
+For day2 cluster operations:
+| Action         | Authorization Access                                                                           |
+|----------------|------------------------------------------------------------------------------------------------|
+| Add Host       | Allowed for every user who has update access for the day1's cluster subscription               |
+
+Filtering for you own clusters:
+Get Cluster call shall have an additional query parameter which filters the cluster by owners only:
+`GET /v2/clusters?owneronly=true`
+
+## Authorization Flow
+![authorization flow](./auth_flow.png)
+
+
+Upon receiving a request from a client, the authorization layer (acting as a reverse proxy) issues an access_review and capability_review requests to ACM to establish the user/agent role:
+
+| Role          | Description                                | Criteria (ACM)                                     |
+|---------------|--------------------------------------------|----------------------------------------------------|
+| ReadOnlyAdmin | Customer support personas                  | Has bare_metal_installer_admin capability          |
+| User          | Users/Agents allowed to create AI clusters | Has create permission on BareMetalCluster resource |
+
+The calls to ACM map to internal static roles, and the authorization layer verifies that the requested API is allowed for that particular role.
+
+If the call was issued by a `User`, there is another dynamic authorization phase, where the authorizer verifies that the user has ownership over the requested resource.
+
+Currently, `ownership` means that the user created the resource (cluster or infraEnv).
+
+## Suggested Changes
+To establish a tenancy-based authorization, some changes should apply to the dynamic part of the flow:
+- 'Ownership' is now defined on an organization level (a user is allowed access if they belong to the same organization as the resource's creator).
+- For Update and Delete of cluster-related APIs, the authorizer shall issue an additional query to ACM to check if the user is allowed to change the resource.
+- If the API is related to a cluster-bound InfraEnv, the access permission is established by the bound cluster as described above. 
+- If the API is related to a non-bound InfraEnv, ownership for reading access is established on the organization level. Update and Delete operations are allowed only to the resource creator.
+
+On the API level we offer an extension to `ListClusters` API as described above to allow filtering by owner.
+
+The entire change will be included in a feature flag to allow a fallback to the current behavior.
+
+## UI Considerations
```

Comment:
wdym? Are'nt we doing all this effort also in order to allow consolidation of the 2 views?

Reply:
@avishayt

Reply:
Yes, once this work is done the UI can combine day1 with OCM.